crates/sncast/src/lib.rs

```diff
@@ -330,18 +331,58 @@ pub fn get_block_id(value: &str) -> Result<BlockId> {
     }
 }
 
+#[derive(Debug)]
+pub struct ErrorData {
+    pub data: String,
+}
+
+impl ErrorData {
+    #[must_use]
+    pub fn new(data: String) -> Self {
+        ErrorData { data }
+    }
+}
+
+impl From<ContractErrorData> for ErrorData {
+    fn from(value: ContractErrorData) -> Self {
+        ErrorData {
+            data: value.revert_error,
+        }
+    }
+}
+
+#[derive(Error, Debug)]
+pub enum TransactionError {
+    #[error("Transaction has been rejected")]
+    Rejected,
+    #[error("Transaction has been reverted = {}", .0.data)]
+    Reverted(ErrorData),
+}
+
+#[derive(Error, Debug)]
+pub enum WaitForTransactionError {
```

Comment:
`AwaitedTransactionError` could be a better name. Current name suggests that it is a selection for what error program should wait.